clients/src/main/resources/common/message/LeaderAndIsrRequest.json

```diff
@@ -78,8 +78,8 @@
         "about": "The leader epoch." },
       { "name": "Isr", "type": "[]int32", "versions": "0+", "entityType": "brokerId",
         "about": "The in-sync replica IDs." },
-      { "name": "ZkVersion", "type": "int32", "versions": "0+",
-        "about": "The ZooKeeper version." },
+      { "name": "PartitionEpoch", "type": "int32", "versions": "0+",
+        "about": "The current epoch for the partition for KRaft controllers. The current ZK version for the legacy controllers." },
```

Comment:
It seems LeaderAndIsrRequest is only used in ZkController and will be removed after KIP-500, will we use it in KRaft mode?

Reply:
That's right. However, we use `LeaderAndIsrPartitionState` in many places internally in the ReplicaManager and in the Partition so it makes sense to unify the naming at this level, I think.

Reply:
I do agree that mentioning KRaft is misleading. How about the following `The current epoch for the partition. The epoch is a monotonically increasing value which is incremented after every partition change.` for the about field?

Reply:
We can add the same comment to `kafka.api.LeaderAndIsr`.

Reply:
Maybe something like this:
```
The current epoch for the partition. The epoch is a monotonically increasing value which is incremented after every partition change. (Since the LeaderAndIsr request is only used by the legacy controller, this corresponds to the zkVersion)
```